### DISCUSSION TITLE
Validate a subset of the schema?

### DISCUSSION THREAD

**Discussion by synchronizing** [python-jsonschema/jsonschema#942] [Q&A] [answered]
I was wondering if it's possible to validate a subset of the schema? That is, say the we have the following schema:

```json
{
    "type": "object",
    "properties": {
        "database": {
            "type": "object",
            "properties": {
                "username": {"type": "string"},
                "password": {"type": "string"}
            },
            "required": ["username", "password"]
        },
        "brands": {"type": "array"},
        "version": {"type": "number"}
    },
    "required": ["database", "brands", "version"]
}
```

And I would like to check only `username`:

```json
{
    "database": {
        "username": "local_username",
    },
}
```

Is this possible? I looked through the documentation, issues, and discussions, but did not find anything. Thanks!

**Comment by gazpachoking** [comment, marked as answer]
You could run the validation, but then ignore any validation errors coming from the 'required' keyword.

**Reply by synchronizing** [reply]
Not a bad idea at all - did not think of this. Will give this a shot and report back in, thank you!

**Reply by synchronizing** [reply]
Worked as expected. Thank you for the suggestion!
